pkg/lib/ratelimit/error.go

```diff
@@ -2,18 +2,19 @@ package ratelimit
 
 import (
 	"github.com/authgear/authgear-server/pkg/api/apierrors"
+	"github.com/authgear/authgear-server/pkg/util/errorutil"
 )
 
 var ErrUsageLimitExceeded = apierrors.ServiceUnavailable.WithReason("UsageLimitExceeded").
 	New("usage limit exceeded")
 
+var errTooManyRequests = apierrors.TooManyRequest.WithReason("RateLimited").New("request rate limited")
+
 func ErrTooManyRequestsFrom(bucket Bucket) error {
-	RateLimited := apierrors.TooManyRequest.WithReason("RateLimited")
-	errMsg := "request rate limited"
 	if bucket.Name == "" {
-		return RateLimited.New(errMsg)
+		return errTooManyRequests
 	} else {
-		return RateLimited.NewWithInfo(errMsg, apierrors.Details{
+		return apierrors.AddInfo(errTooManyRequests, errorutil.Details{
```

Comment:
First of all, we cannot modify a global variable in a request goroutine. This means there could be multiple request goroutines race to modify the global variable with different value.

Why do we need the same instance in the first place?

Reply:
Currently we compare errors using `errors.Is()` during error handling. This function will return false if the two arguments are different error instances: `errors.Is(errors.New("test"), errors.New("test")) == false`, which is the case when comparing errors from `BucketError()`

Reply:
https://pkg.go.dev/errors#Is An error can implement Is() to define equality. @carmenlau I suggest we implement Is() as follows. Thoughts?

```golang
type skyerr struct {
  kind Kind
}

func (e *skyerr) Is(err error) bool {
  apiError := AsAPIError(err)
  if apiError == nil {
    return false
  }
  return e.Kind == apiError.Kind
}

func (e *APIError) Is(err error) bool {
  apiError := AsAPIError(err)
  if apiError == nil {
    return false
  }
  return e.Kind == apiError.Kind
}
```

Reply:
1. Move `RateLimited := apierrors.TooManyRequest.WithReason("RateLimited")` to a module variable
2. Use apierrors.IsKind instead errors.Is in pkg/lib/interaction/nodes/utils_authn.go:87 and pkg/lib/interaction/nodes/verify_identity.go:124